[ZEPPELIN-2298] Remove -Ppyspark build profile

AhyoungRyu opened the pull request:
### What is this PR for?
Currently users who build Zeppelin from source need to include `-Ppyspark` to use `%pyspark` with embedded local Spark. But it's quite inconvenient to write this build profile every time we build i think. So I removed `-Ppyspark` and make pyspark related libraries automatically downloaded when we build Zeppelin. 

### What type of PR is it?
Improvement

### Todos
* [x] - remove the rest of `-Ppyspark` build profile in `dev/create_release.sh`, `dev/publish_release.sh`, and `docs/install/build.md` after getting feedback

### What is the Jira issue?
[ZEPPELIN-2298](https://issues.apache.org/jira/browse/ZEPPELIN-2298)

### How should this be tested?
1. Apply this patch
2. Build source with below command
```
mvn clean package -DskipTests -pl 'zeppelin-interpreter, zeppelin-zengine, zeppelin-server, zeppelin-display, spark, spark-dependencies'

```
Aftr this step, there will be `pyspark` dir under `ZEPPELIN_HOME/interpreter/spark`. Before this PR, only `dep` dir and `zeppelin-spark_2.10-0.8.0-SNAPSHOT.jar` is generated without `-Ppyspark` build profile.

4. Restart Zeppelin. To make sure, run any python code e.g. 
```
%pyspark
print("Hello "+z.input("name"))
``` 
It should be run successfully without any error

### Screenshots (if appropriate)
 tl;dr Without `-Ppyspark` profile
 - Before
<img width="856" alt="screen shot 2017-04-02 at 2 50 57 pm" src="https://cloud.githubusercontent.com/assets/10060731/24584778/0e8ec6b0-17b4-11e7-9f0d-f2599fd7bd63.png">
 
 - After
<img width="893" alt="screen shot 2017-04-02 at 2 28 21 pm" src="https://cloud.githubusercontent.com/assets/10060731/24584779/10b7ed68-17b4-11e7-90d4-aa95eb9bba2d.png">

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

I want to include `SparkR` by default(= remove `-PsparkR` build profile) like this as a next step. I want to ask how Zeppelin community think about this.

felixcheung commented on spark-dependencies/pom.xml:
```diff
@@ -1045,6 +982,63 @@
+              <target>
+                <delete dir="../interpreter/spark/pyspark"/>
+                <copy todir="../interpreter/spark/pyspark"
+                      file="${project.build.directory}/${spark.archive}/python/lib/py4j-${py4j.version}-src.zip"/>
```
this `py4j-${py4j.version}-src.zip` ties to the spark version - perhaps we should change it to spark.py4j.version?

AhyoungRyu replied:
Thanks for pointing it out. Right it makes sense. 
I changed all `py4j.version` in `spark/pom.xml` and `spark-dependencies/pom.xml` to `spark.py4j.version`: ce431584dc77408104b01d0fb46db324021298f4. And also changed `py4j.version`  in `python/pom.xml` to `python.py4j.version`: cee1e874973f625211c6da5f4c643bca74ba48fb

AhyoungRyu commented:
@felixcheung Thanks for pointing it out. Right it makes sense. 
I changed all `py4j.version` in `spark/pom.xml` and `spark-dependencies/pom.xml` to `spark.py4j.version`: ce431584dc77408104b01d0fb46db324021298f4. And also changed `py4j.version`  in `python/pom.xml` to `python.py4j.version`: cee1e874973f625211c6da5f4c643bca74ba48fb

jongyoul commented:
@AhyoungRyu I think you can remove '-Pyarn', too. Can you handle it? You'd better handle it with another PR.

AhyoungRyu commented:
@jongyoul Yup, will open a PR for that soon. 
@Leemoonsoo Thanks. Updated again :)

Leemoonsoo commented:
LGTM

AhyoungRyu commented:
Merge into master if there are no more comments!
